configure.ac

```diff
@@ -1457,9 +1455,12 @@ dnl Check for libnatpmp (optional).
 if test "$use_natpmp" != "no"; then
   TEMP_CPPFLAGS="$CPPFLAGS"
   CPPFLAGS="$CPPFLAGS $NATPMP_CPPFLAGS"
-  AC_CHECK_HEADERS([natpmp.h],
-                   [AC_CHECK_LIB([natpmp], [initnatpmp], [NATPMP_LIBS="$NATPMP_LIBS -lnatpmp"], [have_natpmp=no], [$NATPMP_LIBS])],
-                   [have_natpmp=no])
+  AC_CHECK_HEADERS([natpmp.h], [], [have_natpmp=no])
+
+  if test "$have_natpmp" != "no"; then
+    AC_CHECK_LIB([natpmp], [initnatpmp], [NATPMP_LIBS="$NATPMP_LIBS -lnatpmp"], [], [$NATPMP_LIBS])
```

Comment:
The `AC_CHECK_LIB` macro still have to set `have_natpmp=no` if the library not found, as it is used later: https://github.com/bitcoin/bitcoin/blob/9a823890be9a6b4edfe6f32890f61aba9605dbbc/configure.ac#L1750

Same for `have_miniupnpc`: https://github.com/bitcoin/bitcoin/blob/9a823890be9a6b4edfe6f32890f61aba9605dbbc/configure.ac#L1729

Reply:
I think this is covering an edge case that is very unlikely to happen, but I'll add it in any case.